test/NonFungibleCollectable.test.js

```diff
@@ -0,0 +1,111 @@
+      signatureObject = computeSignature(_firstTokenId, recipientAddress, issuingWallet, web3)
+      solHashResult = await token.hasher(_firstTokenId, {from: recipientAddress})
+    })
+
+    it('should hash the user address and tokenId the same in javascript as in solidity', async () => {
+      signatureObject.hash.should.be.equal(solHashResult)
+    })
+
+    it('user should be able to claim an a token from the collectableCouponIssuer if signature is correct', async () => {
+      const {hash, v, r, s} = signatureObject
+      const issueTx = await token.claimCollectableWithCoupon(_firstTokenId, hash, v, r, s, {from: recipientAddress})
+      const newOwner = await token.ownerOf(_firstTokenId)
+      newOwner.should.be.equal(recipientAddress)
+    })
+
+    describe('when the user tries to use someone elses coupon/signature', () => {
+      it('reverts', async () => {
+        const {hash, v, r, s} = signatureObject
+        await assertRevert(token.claimCollectableWithCoupon(_firstTokenId, hash, v, r, s, {from: attackerAddress}))
+        const newOwner = await token.ownerOf(_firstTokenId)
+        newOwner.should.be.equal(issuingWallet)
+        const balance2 = await token.balanceOf(attackerAddress)
+        balance2.should.be.bignumber.equal(0)
+      })
+    })
+
+    describe('when the user tries to claim a token that was not assigned to them via a valid coupon', () => {
+      it('reverts', async () => {
+        const {hash, v, r, s} = signatureObject
+        await assertRevert(token.claimCollectableWithCoupon(_secondTokenId, hash, v, r, s, {from: recipientAddress}))
+        const newOwner = await token.ownerOf(_firstTokenId)
+        newOwner.should.be.equal(issuingWallet)
+        const balance2 = await token.balanceOf(attackerAddress)
+        balance2.should.be.bignumber.equal(0)
+      })
+    })
+
+    describe('when the attacker tries to use a signature generated by a party other than the `collectableCouponIssuer`', () => {
+      it('reverts', async () => {
+        const {hash, v, r, s} = computeSignature(_firstTokenId, recipientAddress, attackerAddress, web3)
+        await assertRevert(token.claimCollectableWithCoupon(_firstTokenId, hash, v, r, s, {from: recipientAddress}))
+        const newOwner = await token.ownerOf(_firstTokenId)
+        newOwner.should.be.equal(issuingWallet)
+        const balance2 = await token.balanceOf(attackerAddress)
+        balance2.should.be.bignumber.equal(0)
+      })
+    })
+
+    describe('when the attacker tries to claim a token not owned by `collectableCouponIssuer`', () => {
+      it('reverts', async () => {
+        await token.transfer(recipientAddress, _firstTokenId, { from: issuingWallet });
+        const newOwner = await token.ownerOf(_firstTokenId)
+        newOwner.should.be.equal(recipientAddress)
+
+        const {hash, v, r, s} = signatureObject
+        await assertRevert(token.claimCollectableWithCoupon(_firstTokenId, hash, v, r, s, {from: recipientAddress}))
```

Comment:
This test would make more sense to me if it were `{from: attackerAddress}`. As it currently stands, my understanding is that this reverts because it calls NonFungibleCollectable at L40 and reverts due to NonFungibleToken at L235 (i.e. you have transferred the token to the `recipientAddress` and trying to claim a token you own already reverts...

Reply:
Ok, sure, I can change that :)